### PR TITLE
Fix assertion whitelist == nullptr for rocksdb_read_free_rpl_tables

### PR DESCRIPTION
**Pull request by hermanlee**
Summary:

It is valid for NULL to be passed in as an argument for
read_free_rpl_tables and skip_unique_check. Handle NULL arguments in
these functions by treating it as the default value.

Addresses https://github.com/facebook/mysql-5.6/issues/844

Test Plan:

mtr

### CODE REVIEW

File: mysql-test/suite/binlog/t/binlog_rows_query_with_filter.test [hunk 1]
```diff
@@ -38,9 +38,9 @@ let $MYSQLD_DATADIR= `select @@datadir`;
 flush logs;
 
 echo "There should be eight rows_query events in the binlog for d1 and d2 and none for test";
-exec $MYSQL_BINLOG -vvv --skip_empty_trans --skip-gtid --database=d1 $MYSQLD_DATADIR/$master_binlog | grep Rows_query | wc -l;
-exec $MYSQL_BINLOG -vvv --skip_empty_trans --skip-gtid --database=d2 $MYSQLD_DATADIR/$master_binlog | grep Rows_query | wc -l;
-exec $MYSQL_BINLOG -vvv --skip_empty_trans --skip-gtid --database=test $MYSQLD_DATADIR/$master_binlog | grep Rows_query | wc -l;
+exec $MYSQL_BINLOG -vvv --skip_empty_trans --skip-gtids --database=d1 $MYSQLD_DATADIR/$master_binlog | grep Rows_query | wc -l;
```
Comment (lth) [inline mysql-test/suite/binlog/t/binlog_rows_query_with_filter.test]: Are these changes intentional?

**Review by lth — COMMENTED** [review]
Code changes look good to me. Looks like `rocksdb_sys_vars.rocksdb_skip_unique_check_tables_basic` is failing though.

File: mysql-test/suite/binlog/t/binlog_rows_query_with_filter.test (hunk 1 above)
Comment (hermanlee) [reply mysql-test/suite/binlog/t/binlog_rows_query_with_filter.test]: No, they seem to be left over from some previous testing where --skip-gtid is deprecated in 8.0. I'll remove it.

**Comment by george-lorch** [comment]
@hermanlee gentle ping

**Comment by hermanlee** [comment]
@georgelorchpercona, sorry, updating the test case now.

**Review by lth — APPROVED** [review]
I think we need clang-format.
